### PR TITLE
Add environment, logger and refactor

### PR DESCRIPTION
**Pull request by Daanra**
This pull request introduces two significant new concepts:

- Environments
- Logger

**Environments**
I added three environment: **prod**uction, **dev**elopment and **test**. The different environments all come with custom configuration files. The configuration file can be used as a global variable that contains all variables that are installation dependent and makes sure nothing is hardcoded. I have implemented the general concept but it is not entirely complete yet. We still need to add quite a few of our 'secrets' (e.g. Google Calendar API session key) to the config files. This will be done in a future PR.

**Logger**
I added a logger module which should be used instead of `console.log()` on the backend. The logger allows us to have a certain hierarchy in the printed messages and hide some messages in some situations. For example, we might only want to print debugging messages when we run in development mode. It also allows us to act upon errors in a uniform way.

I also refactored quite a lot of code that I encountered on the way and fixed some bugs here and there.

Sorry I make my PR's too big.

### CODE REVIEW

File: config/lib/authorisation.js [hunk 1]
```diff
@@ -1,17 +1,16 @@
-const UserRights = require("../public/dist/js/userRights.js");
-const mongoose = require('mongoose');
+const UserRights = require(process.cwd() + "/public/dist/js/userRights.js");
```
Comment (NikolasHadjipanayi) [inline config/lib/authorisation.js]: proccess.cwd is indeed a much nicer way to address the directory. :+1:

File: config/lib/authorisation.js [hunk 2]
```diff
@@ -24,28 +23,29 @@ exports.isLoggedIn = function (req, res, next) {
...
 	});
 	return hasRights;
-}
+};
 
 /**
  * A custom function that checks whether the user is authorised to perform a request (/access a webpage).
  * It is build on top of the Passport library @see {@link http://passportjs.org/}
+ * If we are in the testing environment no authorisation is performed.
  *
```
Comment (NikolasHadjipanayi) [inline config/lib/authorisation.js]: maybe add development mode in addition to testing here?

File: controllers/API/announcement.js [hunk 3]
```diff
@@ -15,28 +16,36 @@ router.put('/API/announcement', data.isAuthorised("ALTER_ANNOUNCEMENTS"), functi
 			description: req.param('description')
 		}
 	}, function (err, user) {
-		if (err) throw err;
-		res.send(200);
+		if (err) {
+			logger.warning('Unable to edit announcement.\n' + err);
+			res.send(400);
```
Comment (NikolasHadjipanayi) [inline controllers/API/announcement.js]: According to standard convention res code 400 is indeed more appropriate for error responses am going to change mine as well and update the pdf Charles send to reflect this.

File: config/lib/authorisation.js (hunk 2 above)
Comment (Daanra) [reply config/lib/authorisation.js]: I thought about that as well, but that would make us unable to test whether the user rights and login are properly working in the development environment. So I am still a bit doubtful about adding it.

Comment (Daanra) [reply config/lib/authorisation.js]: I thought about that, but that would mean we can not actually test whether the authorisation and authentication system is working properly in development mode. In testing mode it allows us to easily test the API without having to set user rights and all that. So I am not entirely sure whether I should add that. Maybe a better alternative is to add the option to set some environment variable which allows you to skip login and user rights.

**Pull request merged**